pulp_container/app/registry_api.py

```diff
@@ -473,6 +477,134 @@ def get_queryset(self, *args, **kwargs):
+                raise ParseError(detail=f"Unsupported {key}.")
+
+        if "org.flatpak.ref" not in req_label_exists:
+            raise ParseError(detail="Missing label:org.flatpak.ref:exists=1.")
+
+        distributions = models.ContainerDistribution.objects.filter(private=False).only("base_path")
+
+        if req_repositories:
+            distributions = distributions.filter(base_path__in=req_repositories)
+
+        results = []
+        for distribution in distributions:
+            images = []
+            if distribution.repository:
+                repository_version = distribution.repository.latest_version()
+            else:
+                repository_version = distribution.repository_version
+            tags = models.Tag.objects.select_related("tagged_manifest").filter(
+                pk__in=repository_version.content
+            )
+            if req_tags:
+                tags = tags.filter(name__in=req_tags)
+            manifests = {}  # mapping manifests to sets of encountered tag names
+            for tag in tags:
+                self.recurse_through_manifest_lists(
+                    tag.name, tag.tagged_manifest, req_oss, req_architectures, manifests
+                )
+            for manifest, tagged in manifests.items():
+                with storage.open(manifest.config_blob._artifacts.get().file.name) as file:
+                    raw_data = file.read()
+                config_data = json.loads(raw_data)
+                labels = config_data.get("config", {}).get("Labels")
+                if not labels:
+                    continue
+                if any(label not in labels.keys() for label in req_label_exists):
+                    continue
+                os = config_data["os"]
+                if req_oss and os not in req_oss:
+                    continue
+                architecture = config_data["architecture"]
+                if req_architectures and architecture not in req_architectures:
+                    continue
+                if any(
+                    labels.get(label) not in values for label, values in req_label_values.items()
+                ):
+                    continue
+                images.append(
+                    {
+                        "Tags": tagged,
+                        "Digest": manifest.digest,
+                        "MediaType": manifest.media_type,
+                        "OS": os,
+                        "Architecture": architecture,
+                        "Labels": labels,
+                    }
+                )
+            if images:
+                results.append({"Name": distribution.base_path, "Images": images})
+
+        return Response(data={"Registry": settings.CONTENT_ORIGIN, "Results": results})
+
+
+class FlatpakIndexStaticView(FlatpakIndexDynamicView):
+    """
+    Handles requests to the /index/static endpoint
+    """
+
+    @FlatpakIndexStaticCache(expires_ttl=60 * 60 * 8)  # somewhat arbitrary 8h expiration time
```

Comment:
As pointed out in previous comment the default cache expiration is 10mins, or the value that is explicitly configured in Pulp cache settings `EXPIRES_TTL`.
With this hardcoded approach any cache expiration setting will be ignored. This is not very explicit. I am suggesting a) follow the regular setting values or b) if we want to keep this to 8h exp. time then can we please at least document this, and also mention that this is not *changeable value* either. c) introduce a separate flatpak_cache_expires_ttl setting.

@owtaylor do you have any insights or suggestions of what would be a reasonable caching value for the static index?

Reply:
8h seems far too long to me - since it doesn't look like the cache is ever invalidated. It would be very annoying if changes to the content weren't seen for 8 hours. Even 10m seems a bit long, but I don't see a compelling reason to override the overall pulp settings.

Various improvements here come to mind:
 * Set a similar value as max-age on the response (or set no-cache). Flatpak defaults to 30m if no max-age is set - so a server admin might be confused that clients don't see changes for many times a configured EXPIRES_TTL.
 * Compute an ETag as hash(contents), so that in the typical case where the index is constant for days, we don't keep sending contents. Doesn't matter much for small numbers of Flatpak distributions in Pulp, which is the use case we are planning for - would matter a lot for large indexes.

I don't think either is necessary to land this PR.

Reply:
Okay, I propose we leave it the way it was (default pulp): https://github.com/pulp/pulp_container/pull/1301#discussion_r1260332898

Reply:
> 8h seems far too long to me - since it doesn't look like the cache is ever invalidated.

Just to clarify:  The cache _is_ invalidated, see `invalidate_flatpak_index_cache` in `pulp_container/app/models.py`.  (Hence my earlier comment at <https://github.com/pulp/pulp_container/pull/1301#discussion_r1261166968>:  "And my understanding would be that there doesn't need to be any expiration anyway if the invalidation logic is correct?")

And I'm fine with reverting this back to just use the default again.

Reply:
> And I'm fine with reverting this back to just use the default again.

done in c384e8a3ddb22258c3cc70760b8520322560e57c